examples/vision/xray_classification_with_tpus.py

```diff
@@ -0,0 +1,476 @@
+    tf.tpu.experimental.initialize_tpu_system(tpu)
+    strategy = tf.distribute.experimental.TPUStrategy(tpu)
+except:
+    strategy = tf.distribute.get_strategy()
+print("Number of replicas:", strategy.num_replicas_in_sync)
+
+print(tf.__version__)
+
+
+"""
+We need a Google Cloud link to our data to load the data using a TPU. While we're at it,
+we instantiate constant variables. It is generally better practice to use constant
+variables instead of hard-coding numbers.
+"""
+
+AUTOTUNE = tf.data.experimental.AUTOTUNE
+GCS_PATH = "gs://kds-63fdbf6403ae1d5e971e6cdbd4104b02e914cf6e90a178bd0e3990a8"
+BATCH_SIZE = 16 * strategy.num_replicas_in_sync
+IMAGE_SIZE = [180, 180]
+
+"""
+## Load the data
+
+The Chest X-ray data we are using from
+[*Cell*](https://www.kaggle.com/paultimothymooney/chest-xray-pneumonia) divides the data
+into train, val, and test files. There are only 16 files in the validation folder, and we
+would prefer to have a less extreme division between the training and the validation set.
+We will append the validation files and create a new split that resembes the standard
+80:20 division instead.
+"""
+
+filenames = tf.io.gfile.glob(str(GCS_PATH + "/chest_xray/train/*/*"))
+filenames.extend(tf.io.gfile.glob(str(GCS_PATH + "/chest_xray/val/*/*")))
+
+train_filenames, val_filenames = train_test_split(filenames, test_size=0.2)
+
+"""
+Run the following cell to see how many healthy/normal chest X-rays we have and how many
+pneumonia chest X-rays we have.
+"""
+
+COUNT_NORMAL = len([filename for filename in train_filenames if "NORMAL" in filename])
+print("Normal images count in training set: " + str(COUNT_NORMAL))
+
+COUNT_PNEUMONIA = len(
+    [filename for filename in train_filenames if "PNEUMONIA" in filename]
+)
+print("Pneumonia images count in training set: " + str(COUNT_PNEUMONIA))
+
+"""
+Notice that the there are way more images that are classified as pneumonia than normal.
+This shows that we have a imbalance in our data. We will correct for this imbalance later
+on in our notebook.
+"""
+
+train_list_ds = tf.data.Dataset.from_tensor_slices(train_filenames)
+val_list_ds = tf.data.Dataset.from_tensor_slices(val_filenames)
+
+for f in train_list_ds.take(5):
+    print(f.numpy())
+
+"""
+Run the following cell to see how many images we have in our training dataset and how
+many images we have in our validation set. Verify that the ratio of images is 80:20.
+"""
+
+TRAIN_IMG_COUNT = tf.data.experimental.cardinality(train_list_ds).numpy()
+print("Training images count: " + str(TRAIN_IMG_COUNT))
+
+VAL_IMG_COUNT = tf.data.experimental.cardinality(val_list_ds).numpy()
+print("Validating images count: " + str(VAL_IMG_COUNT))
+
+"""
+As expected, we have two labels for our images.
+"""
+
+CLASS_NAMES = np.array(
+    [
+        str(tf.strings.split(item, os.path.sep)[-1].numpy())[2:-1]
+        for item in tf.io.gfile.glob(str(GCS_PATH + "/chest_xray/train/*"))
+    ]
+)
+CLASS_NAMES
```

Comment:
In general, this file works both as a standalone Python script and as a notebook, so if you want to print something, use `print(...)`, e.g. here `print('Class names: %s' % (CLASS_NAMES,))`